### DISCUSSION TITLE
interested in a sell-gear-recommender?

### DISCUSSION THREAD

**Discussion by kfishkin** [LukeCroteau/RaidExtractor#42] [Show and tell]
hi all, this is a bit off-topic but I hope this is ok.

I wanted to teach myself React, so I wrote a program that takes the JSON from the RaidExtractor and uses a number of heuristics to suggest gear you can sell.  Here's a sample:
![balrog_snap](https://user-images.githubusercontent.com/6618823/111918276-8e4f4880-8a41-11eb-9b99-c7b59e595960.PNG)

Three questions for y'all:

- anyone interested in it? I'd be happy to put it on github
- Similarly, i'd be happy to host it somewhere on some web page, if someone could point me to a good place for free or minimal hosting of a React app.
-  any suggestions for other heuristics? The ones I have I got from various youtubers like HellHades, MurderInc, and Chosen. They are presently recommendations to sell:

1. Atk% gloves
2. Epics or legos without a speed substat (this one seems harsh to me)
3. epic non-speed boots
4. gloves of Crit Damage with no SPD or CR substat
5. Defensive boots without a Speed substat
6. Amulets of ATK
7. Defensive Rings w/o 2 good substats
8. Defensive Rings w/o Defensive substats
9. Non-Lego rings w/2 bad substats
10. Under 5 stars
11. Bottom row with flat HP/ATK/DEF main stat.

thanks, sorry if this isn't appropriate here.

**Comment by cmarabate** [comment]
> anyone interested in it? I'd be happy to put it on github

This seems great. I also use React and I would love it if you would put it on Github for me to play around with. Please let me know if you do end up putting it on Github. =)

> Similarly, i'd be happy to host it somewhere on some web page, if someone could point me to a good place for free or minimal hosting of a React app.

I would host it on Netlify, it is free as long as you don't exceed 300 build minutes a month.

**Reply by kfishkin** [reply]
cool, glad you're interested. I'll clean it up a bit and put it on github.

**Comment by redafcb10** [comment]
> anyone interested in it? I'd be happy to put it on github

I am very intered in it. I am waiting for the tool from HH but it seems to not be free anytime soon. I will be looking forwards to yours.


> Similarly, i'd be happy to host it somewhere on some web page, if someone could point me to a good place for free or minimal hosting of a React app.

Netify as a previous comment suggested maybe

> > any suggestions for other heuristics? The ones I have I got from various youtubers like HellHades, MurderInc, and Chosen. They are presently recommendations to sell:
> Atk% gloves

Yes, except on speed sets with speed substats
> Epics or legos without a speed substat (this one seems harsh to me)

Very harsh indeed...
> epic non-speed boots
> Defensive boots without a Speed substat

non-speed boots can be fine, if they are HP% or DEF% with Speed substat or ATK% with Speed, Crit% and Crit. D.%
> gloves of Crit Damage with no SPD or CR substat

Seems fine for me
> Amulets of ATK

Yes, to be sold directly
> Defensive Rings w/o 2 good substats
> Defensive Rings w/o Defensive substats
> Non-Lego rings w/2 bad substats

Defensive rings with main stat (flat value) not in the substats as % are to be sold. Example: if HP ring but without HP% substat it should be sold
> Under 5 stars

For sure
> Bottom row with flat HP/ATK/DEF main stat.

For sure, except speed sets with speed substat (might be useful for arena speed lead where you only care about speed)

To be added in my opinion:
Top row RARE with flat stat (HP/ATK/DEF/RES)

**Comment by LukeCroteau** [comment]
> * anyone interested in it? I'd be happy to put it on github
> * Similarly, i'd be happy to host it somewhere on some web page, if someone could point me to a good place for free or minimal hosting of a React app.

Well, I don't think I need to answer the first question too much! Sounds like people would be very excited to see it!

On the 2nd note, If you (or someone else who tosses in a PR to your new Github repo) dockerize it, I would be more than happy to host it for you on the same server that I host the current webapp embedded in the repository. If it ever blows up, we can talk about moving it to a more suitable permanent host down the road.

Thanks!

**Comment by kfishkin** [comment]
ok, thanks for the signs of interest. I'll clean it up a bit, try and get it running on netify, and then put it up. Be gentle :), this is my first dive into React and it was written as a personal project.

@redafcb10 - I have good news for you. Though the wording is confusing, HH's optimiser _is_ available for free. You just have to create an account on his site, and the lowest level account (I think it's called "common") is free. I use the tool regularly.

I'm thinking of calling the tool 'BRAGO', for "Basic Raid Gear Observer", and perhaps a certain notable recent champion ;-).

**Comment by kfishkin** [comment]
Hey everybody, thanks for the guidance. It's now on github at https://github.com/kfishkin/brago

And it's deployed using Netlify! Go to https://laughing-engelbart-62bcb5.netlify.app/ and give it a spin! The 'help' command has some documentation.
![brago](https://user-images.githubusercontent.com/6618823/112259851-358cd500-8c26-11eb-96a2-083121f485e8.png)

**Reply by cmarabate** [reply]
Wow great work, thank you for sharing! I look forward to playing around with it when I get more time.

**Reply by cmarabate** [reply]
I have had some time to play with it and I am very impressed. Here is what I think it still needs to easier to use...

1) I think the JSON should be added to localStorage so I don't have to upload it again every time I refresh.
2) This one would be a lot of work but I think it would be worth it to have the artifacts be displayed with graphics to look just like they do in-game. They are so much easier to find in your in-game inventory to go and sell them that way. Basically exactly how HellHades' Dev has done it lol...
![image](https://user-images.githubusercontent.com/12236062/112775657-8c6e2200-900b-11eb-89dc-22d93c485ad8.png)
But he doesn't offer us a cool way to find artifacts to sell like you have. So if you got your site pulling in the artifact graphics similar to how his optimizer does, I would use it every time I needed to find gear to sell.
3) This last one isn't really a suggestion because I haven't took the time to go through all your Artifact Checks to Run Toggles and compared them to my notes I use when I am doing a gear cleanse but I figured I would share them with you just incase they help at all...
![image](https://user-images.githubusercontent.com/12236062/112776153-dc99b400-900c-11eb-9a90-642564107732.png)

Keep up the great work and thanks again for sharing it!

**Reply by kfishkin** [reply]
hi Chris, thanks for the feedback.
The artifact pictures are a nice idea, just need the time. It's open-source on github, hint hint :)

My text list is very similar to your, i bet we got 'em from the same places. The problem I had with a lot of them is that they're too vague: "sell X 'unless it's great/godly in other ways'". Well, what defines 'great' or 'godly'? So I just put in the ones that were very clear.

**Comment by kfishkin** [comment]
hi Gang, I just uploaded version 1.02, which includes among other things a 'Champion Detail' page. Mainly useful so you can 'drill down' into the 'total stat' info and see just how it's computed.
